drivers/memc/memc_stm32_sdram.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_fmc_sdram
+
+#include <device.h>
+#include <soc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(memc_stm32_sdram, CONFIG_MEMC_LOG_LEVEL);
+
+/** SDRAM controller register offset. */
+#define SDRAM_OFFSET 0x140U
+
+/** FMC SDRAM controller bank configuration fields. */
+struct memc_stm32_sdram_bank_config {
+	FMC_SDRAM_InitTypeDef init;
+	FMC_SDRAM_TimingTypeDef timing;
+};
+
+/** FMC SDRAM controller configuration fields. */
+struct memc_stm32_sdram_config {
+	FMC_SDRAM_TypeDef *sdram;
+	uint32_t power_up_delay;
+	uint8_t num_auto_refresh;
+	uint16_t mode_register;
+	uint16_t refresh_rate;
+	const struct memc_stm32_sdram_bank_config *banks;
+	size_t banks_len;
+};
+
+static int memc_stm32_sdram_init(const struct device *dev)
+{
+	const struct memc_stm32_sdram_config *config = dev->config;
+
+	SDRAM_HandleTypeDef sdram = { 0 };
+	FMC_SDRAM_CommandTypeDef sdram_cmd = { 0 };
+
+	sdram.Instance = config->sdram;
+
+	for (size_t i = 0U; i < config->banks_len; i++) {
+		sdram.State = HAL_SDRAM_STATE_RESET;
+		memcpy(&sdram.Init, &config->banks[i].init, sizeof(sdram.Init));
+
+		(void)HAL_SDRAM_Init(
+			&sdram,
+			(FMC_SDRAM_TimingTypeDef *)&config->banks[i].timing);
```

Comment:
Note: HAL should be using `const`, files an issue so it will hopefully be fixed in future releases https://github.com/STMicroelectronics/STM32CubeH7/issues/89